specs/ingress/tests.go

```diff
@@ -3,105 +3,145 @@ package ingress
+func testIngress(tc testCase) {
+	// install sample application
+	utils.TestEmojivotoApp()
+	utils.TestEmojivotoInject()
 
-	ginkgo.By("Injecting linkerd into the ingress controller pods")
-	out, err := h.Kubectl("", "get", "-n", utils.NginxNs, "deploy", utils.NginxController, "-o", "yaml")
-	gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("failed to get YAML manifest for deploy/%s: %s", utils.NginxController, utils.Err(err)))
+	h, _ := utils.GetHelperAndConfig()
 
-	out, stderr, err := h.PipeToLinkerdRun(out, "inject", "-")
-	gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("failed to inject: %s", stderr))
+	// install and inject controller
+	for _, url := range tc.controllerURL {
+		out, err := h.Kubectl("",
+			"apply",
+			"-f", url)
+		gomega.Expect(err).Should(gomega.BeNil(),
+			fmt.Sprintf("`kubectl apply` command failed: %s\n%s", out, utils.Err(err)))
+	}
 
-	_, err = h.KubectlApply(out, utils.NginxNs)
-	gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("failed to apply injected manifests: %s", utils.Err(err)))
+	err := h.CheckDeployment(tc.namespace, tc.controllerDeployName, 1)
 
-	err = h.CheckPods(utils.NginxNs, utils.NginxController, 1)
-	gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("failed to verify controller pods: %s", utils.Err(err)))
+	out, err := h.Kubectl("",
+		"get", "-n", tc.namespace,
+		"deploy", tc.controllerDeployName,
+		"-o", "yaml")
+	gomega.Expect(err).Should(gomega.BeNil(),
+		fmt.Sprintf("`kubectl get deploy` command failed: %s\n%s", out, utils.Err(err)))
 
-	ginkgo.By("Verifying if ingress controller pods have been injected")
-	// Wait upto 3mins for proxy container to show up
-	err = utils.CheckProxyContainer(utils.NginxController, utils.NginxNs)
-	gomega.Expect(err).Should(gomega.BeNil(), utils.Err(err))
+	out, stderr, err := h.PipeToLinkerdRun(out, "inject", "-")
+	gomega.Expect(err).Should(gomega.BeNil(),
+		fmt.Sprintf("`linkerd inject` command failed: %s\n%s", out, stderr))
+
+	out, err = h.KubectlApply(out, tc.namespace)
+	gomega.Expect(err).Should(gomega.BeNil(),
+		fmt.Sprintf("`kubectl apply` command failed: %s\n%s", out, utils.Err(err)))
+
+	err = h.CheckDeployment(tc.namespace, tc.controllerDeployName, 1)
+	gomega.Expect(err).Should(gomega.BeNil(),
+		fmt.Sprintf("failed to verify controller pods: %s", utils.Err(err)))
+
+	err = utils.CheckProxyContainer(tc.controllerDeployName, tc.namespace)
+	gomega.Expect(err).Should(gomega.BeNil(),
+		fmt.Sprintf("could not finx proxy container in controller deployment: %s", utils.Err(err)))
+
+	// install ingress resource
+	out, err = h.Kubectl("", "apply", "-f", tc.resourcePath)
+	gomega.Expect(err).Should(gomega.BeNil(),
+		fmt.Sprintf("`kubectl apply` command failed: %s\n%s", out, utils.Err(err)))
+
+	var cmd *exec.Cmd
+	if tc.ingressName != "ambassador" {
+		cmd, err = beginPortForward(tc.namespace, "deploy/"+tc.controllerDeployName)
```

Comment:
Any reasons why we don't want to use the `svc` of the controller here? E.g., nginx-ingress has the `ingress-nginx-controller` service, and traefik has `traefik-ingress-controller`.